src/BTVNanoCommissioning/workflows/BTA_ttbar_producer.py

```diff
@@ -0,0 +1,527 @@
+        chsel = ak.where(
+            criteria["ee"] & (chsel == 0),
+            ak.fill_none(
+                (-11)
+                * (-11)
+                * ak.firsts(pair_ee[idx_ee])["0"].charge
+                * ak.firsts(pair_ee[idx_ee])["1"].charge,
+                0,
+            ),
+            chsel,
+        )
+        basic_vars["ttbar_chan"] = chsel
+
+        #######################
+        #  Electrons & muons  #
+        #######################
+        # the stored electrons and muons, depending on the assigned channel
+        eles_pass = ak.where(
+            criteria["em"],
+            pair_em[idx_em]["0"],
+            ak.where(
+                criteria["mm"],
+                eles[:, :0],
+                ak.where(
+                    criteria["ee"],
+                    ak.concatenate(
+                        [pair_ee[idx_ee]["0"], pair_ee[idx_ee]["1"]], axis=1
+                    ),
+                    eles,
+                ),
+            ),
+        )
+
+        muons_pass = ak.where(
+            criteria["em"],
+            pair_em[idx_em]["1"],
+            ak.where(
+                criteria["mm"],
+                ak.concatenate([pair_mm[idx_mm]["0"], pair_mm[idx_mm]["1"]], axis=1),
+                ak.where(criteria["ee"], muons[:, :0], muons),
+            ),
+        )
+
+        lep_arrays = ak.zip(
+            {
+                "pt": ak.concatenate([eles_pass.pt, muons_pass.pt], axis=1),
+                "eta": ak.concatenate([eles_pass.eta, muons_pass.eta], axis=1),
+                "phi": ak.concatenate([eles_pass.phi, muons_pass.phi], axis=1),
+                "m": ak.concatenate([eles_pass.mass, muons_pass.mass], axis=1),
+                "ch": ak.concatenate([eles_pass.charge, muons_pass.charge], axis=1),
+                "id": ak.concatenate(
+                    [abs(eles_pass.pdgId), abs(muons_pass.pdgId)], axis=1
+                ),
+            }
+        )
+        if not isRealData:
+            lep_arrays["gid"] = ak.fill_none(
+                ak.concatenate(
+                    [eles_pass.matched_gen.pdgId, muons_pass.matched_gen.pdgId],
+                    axis=1,
+                ),
+                0,
+            )
+
+        if not isRealData:
+            ############
+            #  Genlep  #
+            ############
+            # Genlep: same with nominal BTA workflow
+            _fix = lambda x: ak.fill_none(x, 0)
+            is_lep = (
+                lambda p: (abs(_fix(p.pdgId)) == 11)
+                | (abs(_fix(p.pdgId)) == 13)
+                | (abs(_fix(p.pdgId)) == 15)
+            )
+            is_WZ = lambda p: (abs(_fix(p.pdgId)) == 23) | (abs(_fix(p.pdgId)) == 24)
+            is_heavy_hadron = lambda p, pid: (abs(_fix(p.pdgId)) // 100 == pid) | (
+                abs(_fix(p.pdgId)) // 1000 == pid
+            )
+
+            sel = (
+                is_lep(events.GenPart)
+                & ((events.GenPart.statusFlags & (1 << 13)) > 0)
```

Comment:
Suggest to use `coffea` features `events.GenPart.hasFlags["isLastCopy"]` for better understanding instead of bit